### PR TITLE
Vnext directive enhancements

### PR DESCRIPTION
**Pull request by marcvberg**
#### Description of changes

This change adds in the primaryKey, index, hasOne, hasMany, and belongsTo directives into the V2 transformer, all hidden behind the feature flag

#### Description of how you validated changes
Unit testing and some manual testing in the code

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/master/CONTRIBUTING.md#tests)
- [x] Breaking changes to existing customers are released behind a feature flag or major version update
- [x] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts [hunk 1]
```diff
@@ -368,6 +382,60 @@ describe('AppSyncSwiftVisitor', () => {
...
+        author: String
+        book: String
+      }
+    `;
+    const visitorV1 = getVisitor(schemaV1, 'authorBook');
+    const visitorV2 = getVisitorPipelinedTransformer(schemaV2, 'authorBook');
+    const version1Code = visitorV1.generate();
+    const version2Code = visitorV2.generate();
+
+    expect(version1Code).toMatch(version2Code);
```
Comment (AaronZyLee) [inline packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts:409]: The test should also compare the `CodeGenGenerateEnum.metadata` type in addition to code type

**Review by AaronZyLee — CHANGES_REQUESTED** [review]
Some changes are required for swift modelgen

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts (hunk 1 above)
Comment (AaronZyLee) [reply packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts:409]: I suppose the new test will fail due to the first CR

Comment (marcvberg) [reply packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts:409]: Added

File: packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts [hunk 2]
```diff
@@ -441,13 +441,48 @@ export class AppSyncSwiftVisitor<
...
+            directive: directive
+          });
+        });
+      });
+
+      keyDirectives = fieldDirectiveList
+        .filter(directiveObj => directiveObj.directive.name === 'primaryKey' || directiveObj.directive.name === 'index')
+        .map(directiveObj => {
+          switch(directiveObj.directive.name) {
+            case 'primaryKey':
```
[the remark this answers is not in the document]
Comment (marcvberg) [reply packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts:462]: Fixed, just moved the index case above primary key so it falls through to the same code

File: packages/appsync-modelgen-plugin/src/utils/process-belongs-to.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      };
+    }
+    else {
+      throw new Error(
+        `DataStore does not support 1 to 1 connection with both sides of connection as optional field: ${model.name}.${field.name}`,
+      );
+    }
+  }
+  else {
+    if (!field.isList) {
```
[the remark this answers is not in the document]
Comment (marcvberg) [reply packages/appsync-modelgen-plugin/src/utils/process-belongs-to.ts:56]: Wouldn't be, correcting it

File: packages/appsync-modelgen-plugin/src/utils/process-belongs-to.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  if (!field.isList && otherSideField.isList) {
+      //  One to Many
+    return {
+      kind: CodeGenConnectionType.BELONGS_TO,
+      connectedModel: otherSide,
+      isConnectingFieldAutoCreated,
+      targetName: connectionFields[0] || makeConnectionAttributeName(model.name, field.name),
+    };
+  }
+  else if (!field.isList && !otherSideField.isList) {
```
[the remark this answers is not in the document]
Comment (marcvberg) [reply packages/appsync-modelgen-plugin/src/utils/process-belongs-to.ts:40]: Sure I'll switch it so it just does some invalid schema checking first and then returns the response if nothing is wrong

**Comment by marcvberg** [comment]
@AaronZyLee ready for further review

**Review by AaronZyLee — APPROVED** [review]
LGTM! Thanks for the work!

**Pull request merged**